### DISCUSSION TITLE
Sign verity bundle format with an external PKI

### DISCUSSION THREAD

**Discussion by hatmaalla** [rauc/rauc#1046] [Q&A]
Hello,

We are using RAUC 1.8 with streaming capability in our project, which requires a bundle with verity format (or crypt format).
Among the cybersecurity requirements of my project is the ability to sign the firmware bundle (raucb squashfs file in the case of RAUC) with the company specific PKI.
As I can understand from the [documentation](https://rauc.readthedocs.io/en/latest/reference.html#verity-format), with the verity format, we can only sign the manifest with an external PKI, but we can't do the same for the whole package (like the plain format).

My question is whether this possibility can be supported in the future version of RAUC or there are technical constraints due to streaming that blocks this?

**Comment by jluebbe** [comment]
As the manifest contains the verity root hash over the squashfs payload, the signed manifest is used to authenticate the whole update bundle. Note that the bundle file contains both the squashfs, verity hash tree and signed manifest.

Each block downloaded with streaming is authenticated using the verity hash tree (which is authenticated using the root hash).
